File: app/src/main/java/com/tompee/utilities/filldevicespace/view/adapter/MainViewPagerAdapter.java
package com.tompee.utilities.filldevicespace.view.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.tompee.utilities.filldevicespace.R;
import com.tompee.utilities.filldevicespace.view.fragment.AdvanceFillFragment;
import com.tompee.utilities.filldevicespace.view.fragment.CheckStorageFragment;
import com.tompee.utilities.filldevicespace.view.fragment.EasyFillFragment;

public class MainViewPagerAdapter extends FragmentPagerAdapter {
    private static final int PAGE_COUNT = 3;
    private final Context mContext;
    private final EasyFillFragment mEasyFillFragment;
    private final AdvanceFillFragment mAdvanceFillFragment;
    private final CheckStorageFragment mCheckStorageFragment;

    public MainViewPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        mContext = context;
        mEasyFillFragment = EasyFillFragment.getInstance();
        mAdvanceFillFragment = AdvanceFillFragment.getInstance();
        mCheckStorageFragment = CheckStorageFragment.getInstance();
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 1:
                return mAdvanceFillFragment;
            case 2:
                return mCheckStorageFragment;
            default:
        }
        return mEasyFillFragment;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        String name;
        switch (position) {
            case 0:
                name = mContext.getString(R.string.ids_lbl_easy_fill);
                break;
            case 1:
                name = mContext.getString(R.string.ids_lbl_advance_fill);
                break;
            default:
                name = mContext.getString(R.string.ids_lbl_check_storage);
                break;
        }
        return name;
    }
}

File: app/src/main/java/com/tompee/utilities/filldevicespace/feature/main/MainViewPagerAdapter.kt
package com.tompee.utilities.filldevicespace.feature.main

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.tompee.utilities.filldevicespace.R
import com.tompee.utilities.filldevicespace.feature.main.advancefill.AdvanceFillFragment
import com.tompee.utilities.filldevicespace.feature.main.easyfill.EasyFillFragment
import com.tompee.utilities.filldevicespace.feature.main.storage.CheckStorageFragment

class MainViewPagerAdapter(private val context: Context,
                           private val easyFillFragment: EasyFillFragment,
                           private val advanceFillFragment: AdvanceFillFragment,
                           private val checkStorageFragment: CheckStorageFragment,
                           fragmentManager: FragmentManager) : FragmentPagerAdapter(fragmentManager) {
    companion object {
        private const val PAGE_COUNT = 3
    }

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> easyFillFragment
            1 -> advanceFillFragment
            else -> checkStorageFragment
        }
    }

    override fun getCount(): Int = PAGE_COUNT

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> context.getString(R.string.ids_lbl_easy_fill)
            1 -> context.getString(R.string.ids_lbl_advance_fill)
            else -> context.getString(R.string.ids_lbl_check_storage)
        }
    }
}